core/src/main/java/org/apache/iceberg/rest/responses/DropNamespaceResponse.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Represents a REST response to drop a namespace.
+ * <p>
+ * The server may choose to return an error if the namespace is not empty.
+ */
+public class DropNamespaceResponse {
+
+  // For Jackson to properly fail when deserialziing, this needs to be boxed.
+  private Boolean dropped;
+
+  private DropNamespaceResponse(@JsonProperty("dropped") Boolean dropped) {
+    Preconditions.checkNotNull(dropped,
+        "Cannot instantiate a response to dropping a namespace with missing fields");
+    this.dropped = dropped;
+  }
```

Comment:
The way Jackson handles this with the current `visibilityModifiers` based approach for deserialization, things like `{ "dropped": 123 }` still deserialize as though they were `{dropped: true}`.

Presumably anything that's truthy, but boxing `this.dropped` from primitive `boolean` helped somewhat (at least with strings). I'll add more tests.